torch_xla/csrc/tensor_impl.cpp

```diff
@@ -128,6 +130,15 @@ int64_t XLATensorImpl::size(int64_t d) const {
 }
 
 void XLATensorImpl::SetupSizeProperties() {
+  if (getPythonPrinter() && !disablePrinter()) {
```

Comment:
so if I understand correctly, this will ensure that whenever `tensor.size()` is called, we log the python frame and C++ frame if printer is set. I think it is a useful tool but I am not sure if it is a good idea for us to merge the change in this file into the master. I think it make a lot of sense of add the binding and documents, but this part to me feel like a debugging feature for developer.

Unless we are at a stage where we expect user to dump this info and do debugging, I felt like it might be better to leave this in a separate branch?